### PR TITLE
Minimal OGM for Kuzu

### PR DESCRIPTION
**Pull request by whimo**

### CODE REVIEW

File: examples/research_agent/answer_orchestrator.py [hunk 1]
```diff
@@ -43,7 +45,7 @@ def __call__(self) -> Question | None:
             logging.warning("Nothing to answer!")
             return
 
-        return Question.deserialize(self.storage.get_entity(last_question.id))
+        return Question.deserialize(self.storage.get_entity_by_class_and_id(last_question.id))
```
Comment (ZmeiGorynych) [inline examples/research_agent/answer_orchestrator.py:48]: If it's get_entity_by_class_and_id, why do  you pass only one argument? If the argument is a tuple that contains both, then why the long name? 
Also, the additional deserialize step is a hassle - should just be self.storage.get(id) that returns a ready-made object

Comment (ZmeiGorynych) [reply examples/research_agent/answer_orchestrator.py:48]: Also, didn't we want to make the entities the graph returns immutable?

Comment (ZmeiGorynych) [reply examples/research_agent/answer_orchestrator.py:48]: Finally, I sense a philosophical confusion here - we certainly shouldn't expose the implementation details (the fact that the id is class-specific) in the high-level API. We either just treat the whole immutable object as its own ID, or we treat the tuple (class, id) as the ID on the python side, eg exposed by the object via `@property`. So on the Python side, the id should be a single thing, either the object itself or the tuple (class, db_id)

Comment (whimo) [reply examples/research_agent/answer_orchestrator.py:48]: Seems like you misunderstood the purpose of this PR :)
This is only about the OGM, I almost didn't touch other code. This is WIP, will be merged into the scary_refactor branch and I will continue working there. The code you're commenting on is outdated, it was automatically refactored by the IDE in terms of function names but not in terms of logic.

For understanding the OGM interface, it's better to look at test_kuzu_graph_store.py

Comment (whimo) [reply examples/research_agent/answer_orchestrator.py:48]: By the way, your comments reminded me to also make a test on immutability :)

**Pull request merged**